### ISSUE TITLE
Feature requests: Notifications

### ISSUE THREAD

**Issue by vertigo220** [JustOff/cookies-exterminator#7]
I have two requests regarding the windows notifications. The first is to add an option to adjust the duration, to allow it to be adjust so the pop-up doesn't stay, blocking the bottom-right corner of the screen and browser, for so long. The second is an option to not show notifications that state it's suppressing a duplicate notification; showing one notification to say that it's not showing another notification seems pointless and is needlessly distracting.

By the way, I wasn't sure whether to post this here or on mozillazine, since this is where the support link in the "More information" section on AMO leads to, but in the description it lists mozillazine as the support site. I tried to register there but was unable, as it kept saying my answer to the bot-check question was wrong. I also don't care for the fact it's http only and only allows up to a 30-character password, so frankly, I'd just prefer to not bother with it.

**Comment by JustOff** [comment]
Unfortunately the browser does not provide a way to control the notification time, it's just hard-coded. 

In fact, notifications are provided mostly to understand how the extension works and to make sure everything is correct, in all other cases it is better to use the cleanup log. The notification about the suppression of duplicate notifications is an important part of the initial experience and is shown only once for several identical messages in a row. Thus, I'm sorry, but I'm not going to add a separate setting to suppress duplicate messages.

Mozilla is going to soon remove everything but WebExtensions from AMO, so this place is becoming a new CE home, and it's ok to use issues for support requests.

**Comment by vertigo220** [comment]
To your first point, Skip Redirect (https://github.com/sblask/webextension-skip-redirect) has an option to adjust the notification duration between 1 and 10 seconds, and it works perfectly. I understand most people probably turn them off completely after a while, and I might as well, but I do like seeing what's going on, and if I could make them last just a few seconds I would probably leave them on, so I hope you can add it and decide to do so. That said, I understand if it would be a low priority.

As for the duplicate messages, I didn't realize they were shown only for several duplicates, I thought it was a one-for-one thing, so that makes more sense, though I still think it would be nice to be able to just disable duplicates, but it's not really a big deal. Anyways, thanks for the addon. I hope they get their act together and add the necessary APIs so you can port it to WE.

**Comment by JustOff** [comment]
I took a look at how `Skip Redirect` limits the notifications duration, and found that it simply clears all existing notifications after specified timeout. This could be an option, but I need to check if this will not break anything else.

**Comment by vertigo220** [comment]
Thanks!

**Comment by JustOff** [comment]
Please try [cookexterm-2.9.4b1.xpi](https://github.com/JustOff/cookies-exterminator/releases/download/2.9.3/cookexterm-2.9.4b1.xpi).

**Comment by vertigo220** [comment]
Here are my observations:

1) The timer seems to work perfectly! Thank you. I'll continue to use it with a shortened timer to check for issues.
2) With the normal version, using the default notification duration, the notifications last ~19 seconds, whereas you max it out at 10 seconds in the beta. While I doubt anyone would want or need it to display for longer than that, it is possibly someone might actually be upset by the shorter duration.

Unrelated, but noticed when paying closer attention during testing of the timer:

3) The first time it does a clear after installing, it clears multiple things, which doesn't make sense since anything that needed clearing should have already been done by the previous version, whether stable or beta.
4) Looking in my profile\storage\default folder, I see a https+++www.washingtonpost.com^firstPartyDomain=washingtonpost.com folder. This also shows up in the log each time. However, it's not in the white or grey list and it's not open in any tabs. So it appears that it's not being cleaned for some reason.
5) Frequently, after I close a tab and it cleans the remains from that site, if I reopen the tab then close it again, whether right away or after waiting several seconds or reloading it or even navigating in it, CE doesn't clean it again on the subsequent closes. It's as if once it cleans a site it doesn't see that site again for a while.
6) Looking in my cookies.sqlite file, there were several wikipedia entries, even after closing a wikipedia tab and seeing the notification that it was cleaned. But then I reopened the wikipedia tab and checked again and there were no entries, which is opposite of what I would expect. I navigated around wikipedia a bit, checked again, saw there were still no entries, then closed the tab and got a cleaning notification for wikipedia, even though there doesn't seem to have been anything to clean (maybe in IndexDB?). It just seems like CE is randomly cleaning or not cleaning and showing a notification or not, and I want to be sure there's not an issue that's causing it to not work correctly.

**Comment by JustOff** [comment]
I guess you have enabled first-party isolation, but [this mode is not supported by CE](https://github.com/JustOff/cookies-exterminator/issues/5).

**Comment by vertigo220** [comment]
Indeed I have. I take it that explains all the strange behavior I'm seeing? Do you know what exactly doesn't work right when using FPI, i.e. does it just cause the occasional cookie or local storage item to not be deleted, or does it more or less completely break CE, or is it something in between? I'd really hate to have to choose between FPI and CE, since they both provide important functionality.

**Comment by JustOff** [comment]
Unfortunately, it's just untested, and I do not have the opportunity to pay attention to this, because my primary browser simply does not support FPI.

**Comment by JustOff** [comment]
Resolved by [2.9.4](https://github.com/JustOff/cookies-exterminator/releases/tag/2.9.4).
